RubberduckTests/Inspections/AssignedByValParameterInspectionTests.cs

```diff
@@ -97,19 +97,152 @@ public void AssignedByValParameter_ReturnsResult_SomeAssignedByValParams()
         [TestCategory("Inspections")]
         public void AssignedByValParameter_QuickFixWorks()
```

Comment:
That's excellent. Because the identifier is an `UnrestrictedIdentifier`, I'd also add a test with a bracketed name, e.g. `ByVal [bar] As String` to make sure the square brackets aren't throwing off anything.